crates/red_knot_python_semantic/src/util/subscript.rs

```diff
@@ -18,6 +18,7 @@ impl<I, T: DoubleEndedIterator<Item = I>> PythonSubscript for T {
 }
 
 #[cfg(test)]
+#[allow(clippy::redundant_clone)]
```

Comment:
Why are we allowing it for this module? Could you possibly add a comment?

Reply:
It's just tests and removing the clone would make adding a new case more verbose. I think it's very self-explanatory if you remove the allow, and I don't mind if someone's eager to fix the unnecessary calls. It just didn't felt worth my time.